packages/runtime/lib/worker.js

```diff
@@ -34,28 +35,28 @@ globalThis.fetch = undici.fetch
 const config = workerData.config
 
 let loggerConfig = config.server?.logger
-let destination
 
 if (loggerConfig) {
   loggerConfig = { ...loggerConfig }
 } else {
   loggerConfig = {}
 }
 
-/* c8 ignore next 10 */
+let pinoStream = pino.destination(1)
+if (isatty(1)) {
+  pinoStream = pretty()
+}
 if (config.loggingPort) {
-  destination = new MessagePortWritable({
+  delete loggerConfig.transport
+
+  const portStream = new MessagePortWritable({
     metadata: config.loggingMetadata,
     port: config.loggingPort
   })
-  delete loggerConfig.transport
-} else if (!loggerConfig.transport && isatty(1)) {
-  loggerConfig.transport = {
-    target: 'pino-pretty'
-  }
+  pinoStream = pino.multistream([portStream, pinoStream])
```

Comment:
I think youàd need to set the log levels in a slightly different way. We might want to receive also trace and debug messages on the port, and filter them out on the other side.